Tests/StreamChatUITests/UISDKdocumentation/UIView+Annotations.swift

```diff
@@ -0,0 +1,321 @@
+///
+/// This method creates canvas with size of 700x700 which will be used as background for annotating the views subviews. After creating the canvas,
+/// it generates either highlighting given elements or describing property names in the view with pointing lines.
+/// - Parameters:
+///   - view: Parent view which we want to annotate containing subviews to describe
+///   - annotations: Annotation structs containing details which subviews we want to describe and how
+///   - name: Name of the snapshot documentation.
+///   - variants: Variants to snapshot, this is to support dark mode.
+///   - file: File reference which will be passed to snapshot to generate folder if needed.
+/// - Returns: A UIView instance which contains all annotations and possibly description of properties outside the annotated view.
+func generateDocs(
+    for view: UIView,
+    annotations: [Annotation],
+    name: String,
+    variants: [SnapshotVariant],
+    file: StaticString = #file
+) {
+    let container = UIView().withoutAutoresizingMaskConstraints
+    container.backgroundColor = .clear
+    
+    container.addSubview(view.withoutAutoresizingMaskConstraints)
+    view.centerYAnchor.constraint(equalTo: container.centerYAnchor).isActive = true
+    view.centerXAnchor.constraint(equalTo: container.centerXAnchor).isActive = true
+ 
+    container.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 3).isActive = true
+    container.heightAnchor.constraint(equalTo: view.heightAnchor, multiplier: 3).isActive = true
+    
+    container.layoutIfNeeded()
+    for annotation in annotations {
+        // Create labels describing the properties if needed
+        if let position = annotation.descriptionLabelPosition, annotation.isNameIncluded {
+            createLineAndLabelWithPropertyName(
+                for: [annotation.view: position],
+                in: container,
+                parentView: view,
+                lineColor: annotation.lineColor,
+                textColor: annotation.textColor
+            )
+        }
+        
+        // Highlight the given view/container if needed
+        if let highlightColor = annotation.highlightColor {
+            annotation.view.backgroundColor = highlightColor
+        }
+    }
+    AssertSnapshot(container, variants: variants, file: file, function: name)
+}
+
+private func createLineAndLabelWithPropertyName(
+    for subviewsWithPosition: [UIView: DescriptionPosition],
+    in container: UIView,
+    parentView: UIView,
+    lineColor: UIColor,
+    textColor: UIColor?
+) {
+    let lineDistance: CGFloat = 50
+    
+    for (view, position) in subviewsWithPosition {
+        view.layer.borderWidth = 1
+        view.layer.borderColor = lineColor.cgColor
+        
+        let translated = view.convert(view.bounds, to: container)
+        var end: CGPoint
+        
+        let mirror = Mirror(reflecting: parentView)
+        let mirroredView = mirror.children.first { $0.value as? UIView == view }
```

Comment:
Do we really need a mirror here? 🤔 Why don't we just use `parentView.subviews.first { $0 === view }` ? Also, we probably want to use `===` and not `==` ?

Reply:
Nice catch, you are right. 

>  the C++ function is invoked with value containing a pointer to the Swift value

guess I was afraid to use the `identical to` operator 😅 
https://swift.org/blog/how-mirror-works/